controllers/main.go

```diff
@@ -301,66 +303,64 @@ func (controller *MainController) APIAddress(c web.C, r *http.Request) ([]string
+		log.Errorf("generateTicketPurchaseInfo: %v", err)
 		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
 	}
 
-	poolValidateAddress, err := controller.Cfg.StakepooldServers.ValidateAddress(pooladdress)
-	if err != nil {
-		log.Errorf("unable to validate address: %v", err)
-		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
-	}
-	if !poolValidateAddress.IsMine {
-		log.Errorf("unable to validate ismine for pool ticket address: %s",
-			pooladdress.String())
-		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
-	}
+	return nil, codes.OK, "address successfully imported", nil
+}
 
-	poolPubKeyAddr := poolValidateAddress.PubKeyAddr
+// APIPurchaseTicket returns relevant vsp information to enable clients purchase tickets that this vsp can vote on
+func (controller *MainController) APIPurchaseTicket(c web.C, r *http.Request) (*poolapi.PurchaseInfo, codes.Code, string, error) {
+	userPubKeyAddr := r.FormValue("UserPubKeyAddr")
 
-	if _, err = dcrutil.DecodeAddress(poolPubKeyAddr, controller.Cfg.NetParams); err != nil {
-		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
+	if _, err := validateUserPubKeyAddr(userPubKeyAddr, controller.Cfg.NetParams); err != nil {
+		return nil, codes.InvalidArgument, "address error", err
 	}
 
-	createMultiSig, err := controller.Cfg.StakepooldServers.CreateMultisig([]string{poolPubKeyAddr, userPubKeyAddr})
-	if err != nil {
-		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
-	}
+	dbMap := controller.GetDbMap(c)
 
-	// Serialize the redeem script (hex string -> []byte)
-	serializedScript, err := hex.DecodeString(createMultiSig.RedeemScript)
-	if err != nil {
-		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
+	// check if this userPubKeyAddr has been used previously to generate ticket purchase parameters
+	user := models.GetUserByEmail(dbMap, userPubKeyAddr)
+	if user != nil {
+		purchaseInfo := &poolapi.PurchaseInfo{
+			PoolAddress:   user.UserFeeAddr,
+			PoolFees:      controller.Cfg.PoolFees,
+			Script:        user.MultiSigScript,
+			TicketAddress: user.MultiSigAddress,
+			VoteBits:      uint16(user.VoteBits),
+		}
+		return purchaseInfo, codes.OK, "APIPurchaseTicket: purchaseinfo retrieved for existing userPubKeyAddr", nil
 	}
 
-	// Import the redeem script
-	var importedHeight int64
-	importedHeight, err = controller.Cfg.StakepooldServers.ImportNewScript(serializedScript)
-	if err != nil {
-		return nil, codes.Unavailable, "system error", errors.New("unable to process wallet commands")
+	user = &models.User{
+		Username:        userPubKeyAddr,
+		Email:           userPubKeyAddr,
+		EmailVerified:   0,
+		VoteBits:        1,
+		VoteBitsVersion: int64(controller.voteVersion),
```

Comment:
Perhaps it would be best to put in some random password? Just to be safe and sure noone can access this account through http...